include/libopencm3/stm32/f4/dsi.h

```diff
@@ -0,0 +1,814 @@
+/*
+ * STM32F4 DSI Host Defines
+ *
+ * Copyright (C) 2016, Chuck McManis <cmcmanis@mcmanis.com>
+ *
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/stm32/memorymap.h>
+
+#ifndef DSI_H
+#define DSI_H
+
+/*
+ * A couple of helper macros to avoid copy pasta code
+ * For multi-bit fields we can use these macros to create
+ * a unified SET_<reg>_<field> version and a GET_<reg_<field> version.
+ */
+#define DSI_GET(reg, field, x) \
+    (((x) >> DSI_ ## reg ## _ ## field ##_SHIFT) & DSI_ ## reg ##_ ## field ##_MASK)
+
+#define DSI_SET(reg, field, x) \
+    (((x) & DSI_ ## reg ##_## field ##_MASK) << DSI_## reg ##_## field ##_SHIFT)
+
+#define DSI_MASK(reg, field) \
+    ((DSI_ ## reg ##_## field ##_MASK) << DSI_## reg ##_## field ##_SHIFT)
```

Comment:
I would greatly prefer to see this sort of preprocessor magic left in peoples examples or drivers, not in the common headers.  They don't exist anywhere else in the library, and I don't personally want to start spreading them around.  You're not using them anywhere else in this file, and honestly, the usage of the macros is not obvious and there's no documentation.